namer/namer.cc

```diff
@@ -490,6 +515,25 @@ class SymbolFinder {
                 default:
                     return tree;
             }
+        } else if (original.args.size() == 2) {
+            ENFORCE(!methodVisiStack.empty());
+            if (!methodVisiStack.back().has_value()) {
+                return tree;
+            }
+
+            // visibility toggle doesn't look at self methods, only instance methods
+            if (original.fun != core::Names::keepDef()) {
+                return tree;
+            }
+
+            auto recv = ast::cast_tree<ast::ConstantLit>(original.recv);
+            if (recv == nullptr || recv->symbol != core::Symbols::Sorbet_Private_Static()) {
+                return tree;
+            }
+
+            auto &current = methodVisiStack.back();
+            auto methodName = unwrapLiteralToMethodName(ctx, original.args[1]);
+            foundDefs->addModifier(Modifier{current->kind, current->owner, current->loc, current->name, methodName});
```

Comment:
What do you think about adding a method called `withTarget` to `Modifier` that copies everything, but overwrites the target parameter? Then this line would become:
```c++
foundDefs->addModifier(current.withTarget(methodName));
```
As it is now, you have to read through all the args to know that it's copying everything except for the target.